fcomm/src/lib.rs

```diff
@@ -1085,6 +1086,48 @@ mod test {
     use lurk::public_parameters::public_params;
     use lurk::z_data::{from_z_data, to_z_data};
 
+    // You have broken a snapshot test. Unlike round-trip tests, those tests check the actual format of serialized Lurk expressions,
+    // and since you broke one, it's probable that you have changed that format, which will break at least the fcomm examples.
+    // Please read the documentation on snapshot tests `https://insta.rs/docs/quickstart/`, fix the snapshot **AND**
+    // make sure `cargo nextest run --run-ignored all -E 'test(test_make_fcomm_examples)'` passes
+    #[test]
+    fn test_snapshot_serialized_expressions() {
+        let function_source: &str = "(letrec ((secret 12345) (a (lambda (acc x) (let ((acc (+ acc x))) (cons acc (hide secret (a acc))))))) (a 0))";
+        let function_inputs: &str = "(+ 1 2)";
```

Comment:
Assuming this was intended to not only exercise snapshots but also some aspect of the existing examples, then the input here seems mismatched to the function. I think the input was meant to come from `chained-input.lurk` rather than `eval-input.lurk`.

Reply:
We do have a test that exercises the chained input function ([triggered here](https://github.com/lurk-lab/lurk-rs/blob/adb7290fb6ef6256ca48a7d25ef921d8e5dde3d4/fcomm/examples/Makefile#L4), run in Rust through [here](https://github.com/lurk-lab/lurk-rs/blob/master/fcomm/tests/makefile_tests.rs)) as well as one that exercises the eval input ([triggered here](https://github.com/lurk-lab/lurk-rs/blob/adb7290fb6ef6256ca48a7d25ef921d8e5dde3d4/fcomm/examples/Makefile#L2), run in Rust through [here](https://github.com/lurk-lab/lurk-rs/blob/master/fcomm/tests/makefile_tests.rs)).